lib/BuildSystem/LaneBasedExecutionQueue.cpp

```diff
@@ -534,10 +616,98 @@ class LaneBasedExecutionQueue : public BuildExecutionQueue {
             StringRef(buf, numBytes));
       }
 
-      // Close the read end of the pipe.
+      if (readfds[0].revents & POLLIN) {
+        ssize_t numBytes = read(releasePipe[0], buf, sizeof(buf));
+        if (numBytes < 0) {
+          int err = errno;
+          getDelegate().commandProcessHadError(
+               context.job.getForCommand(), handle,
+               Twine("unable to read process release signal (") + strerror(err) + ")");
+          break;
+        }
+
+        if (numBytes == 0) {
+          readfds[0].events = NULL;
+          if (shouldCaptureOutput) {
+            continue;
+          }
+          break;
+        }
+
+        auto msg = StringRef(buf, numBytes).trim();
+        SmallString<16> Vec;
+        if (taskID.toStringRef(Vec).equals_lower(msg)) {
+          // Close the read end of the release pipe.
+          ::close(releasePipe[0]);
+
+          std::thread([
+              this, pid, handle, command=context.job.getForCommand(), outputPipe{outputPipe[0]},
+              completionFn, subprocessInterval{std::move(subprocessInterval)}
+            ]() mutable {
+            if (outputPipe != -1)
```

Comment:
Would prefer to use `shouldCaptureOutput`. Esp. since outputPipe is an array 😛

Reply:
Well, in this case the lambda has just the fd.  Should rename it for clarity.